core/src/solution_submission.rs

```diff
@@ -213,6 +217,19 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
         gas_price_cap: U256,
     ) -> BoxFuture<Result<(), SolutionSubmissionError>> {
         async move {
+            // If the gas price cap is not at least the fast gas price then submitting the solution
+            // is not feasible because it would take too long.
+            match self.gas_price_estimating.estimate_gas_price().await {
+                Ok(gas_price) => {
+                    if gas_price_cap < gas_price.fast {
+                        return Err(SolutionSubmissionError::Benign(format!(
+                            "not submitting solution because gas price cap {} < fast gas price {}",
```

Comment:
Maybe we can reword this error message a bit. I think it should indicate that the solution does not generate enough fees for the current network gas price estimate.